test/MStrategy.test.ts

```diff
@@ -1455,6 +1463,533 @@ contract<MStrategy, DeployOptions, CustomContext>("MStrategy", function () {
+
+                    await subject.swapToTarget(params, []);
+                });
+            });
+        });
+    });
+
+    describe("_rebalancePools", () => {
+        describe("edge cases", () => {
+            describe("when erc20Vault.tvl[i] + moneyVault.tvl[i] > uint256max / 2 or erc20tvl[i] > uint256max / 2", () => {
+                it(`reverts with ${Exceptions.LIMIT_OVERFLOW}`, async () => {
+                    await mint(
+                        "USDC",
+                        this.deployer.address,
+                        ethers.constants.MaxUint256.div(2)
+                    );
+
+                    let {
+                        mockNonfungiblePositionManager,
+                        mockSwapRouter,
+                        mockUniswapV3Pool,
+                    } = await deployMockContracts();
+                    let mockMStrategyFactory = await ethers.getContractFactory(
+                        "MockMStrategy"
+                    );
+                    let mockMStrategy = await mockMStrategyFactory.deploy(
+                        mockNonfungiblePositionManager.address,
+                        mockSwapRouter.address
+                    );
+
+                    const address =
+                        await mockMStrategy.callStatic.createStrategy(
+                            this.params.tokens,
+                            this.params.erc20Vault,
+                            this.params.moneyVault,
+                            this.params.fee,
+                            this.params.admin
+                        );
+                    await mockMStrategy.createStrategy(
+                        this.params.tokens,
+                        this.params.erc20Vault,
+                        this.params.moneyVault,
+                        this.params.fee,
+                        this.params.admin
+                    );
+                    let subject = await ethers.getContractAt(
+                        "MockMStrategy",
+                        address
+                    );
+
+                    await this.usdc
+                        .connect(this.deployer)
+                        .transfer(
+                            this.params.erc20Vault,
+                            ethers.constants.MaxUint256.div(2)
+                        );
+
+                    await expect(
+                        subject
+                            .connect(this.mStrategyAdmin)
+                            .rebalancePools(
+                                this.params.erc20Vault,
+                                this.params.moneyVault,
+                                this.params.tokens,
+                                BigNumber.from(0),
+                                []
+                            )
+                    ).to.be.revertedWith(Exceptions.LIMIT_OVERFLOW);
+                });
+            });
+
+            describe("when targetTokenAmount - erc20Vault.tvl > 0 for both tokens", () => {
```

Comment:
Perhaps it should be `> 0` -> `>= 0` b.o. [comparison](https://github.com/mellow-finance/mellow-vaults/blob/main/contracts/strategies/MStrategy.sol#L283)